src/metabase/api/field.clj

```diff
@@ -98,6 +98,30 @@
                (not (internal-remapping-allowed? base-type new-semantic-type)))
       (t2/delete! Dimension :id old-dim-id))))
 
+(defn- update-nested-fields-on-json-unfolding-change!
+  "If JSON unfolding was enabled for a JSON field, it activates previously synced nested fields from the JSON field.
+   If JSON unfolding was disabled for that field, it inactivates the nested fields from the JSON field.
+   Returns nil."
+  [old-field json_unfolding]
```

Comment:
```suggestion
  [old-field json-unfolding]
```

Reply:
this cond is quite hard to follow IMO, I'd rather have a top-level `(when (not= old-value new-value))` then do the conditional update based on the value of the new value.